modin/core/dataframe/algebra/tree_reduce.py

```diff
@@ -45,11 +49,17 @@ def register(cls, map_function, reduce_function=None, axis=None):
         def caller(query_compiler, *args, **kwargs):
             """Execute TreeReduce function against passed query compiler."""
             _axis = kwargs.get("axis") if axis is None else axis
+
+            new_dtypes = None
+            if compute_dtypes and query_compiler._modin_frame.has_materialized_dtypes:
+                new_dtypes = str(compute_dtypes(query_compiler.dtypes, *args, **kwargs))
+
             return query_compiler.__constructor__(
                 query_compiler._modin_frame.tree_reduce(
                     cls.validate_axis(_axis),
                     lambda x: map_function(x, *args, **kwargs),
                     lambda y: reduce_function(y, *args, **kwargs),
+                    dtypes=new_dtypes,
```

Comment:
Don't you have to rename this argument as well?

Reply:
The type of this variable does not change, so I did not change its name.